arch/x86_64/src/intel64/intel64_lowsetup.c

```diff
@@ -56,24 +71,24 @@
 
 void up_lowsetup(void)
 {
-  /* we should be in long mode at this point*/
+  /* we should be in long mode at this point */
 
   /* GDT is loaded with 64bit GDT  */
 
-  /* Paging is enabled*/
+  /* Paging is enabled */
 
   /* Setup pointers for accessing Page table and GDT in high address */
 
-  pdpt = &pdpt_low + X86_64_LOAD_OFFSET;
-  pd   = &pd_low   + X86_64_LOAD_OFFSET;
-  pt   = &pt_low   + X86_64_LOAD_OFFSET;
+  pdpt = (uint64_t *)((uint64_t)&pdpt_low + X86_64_LOAD_OFFSET);
+  pd   = (uint64_t *)((uint64_t)&pd_low   + X86_64_LOAD_OFFSET);
+  pt   = (uint64_t *)((uint64_t)&pt_low   + X86_64_LOAD_OFFSET);
 
-  ist64     = (struct ist_s *)((void *)&ist64_low       + X86_64_LOAD_OFFSET);
-  gdt64     = (struct gdt_entry_s *)((void *)&gdt64_low + X86_64_LOAD_OFFSET);
+  ist64 = (struct ist_s *)((uint64_t)&ist64_low       + X86_64_LOAD_OFFSET);
+  gdt64 = (struct gdt_entry_s *)((uint64_t)&gdt64_low + X86_64_LOAD_OFFSET);
```

Comment:
do you really need this change? (void *) seems perfectly fine to me.

Reply:
Yes otherwise you will get a alias warning. Also you need to cast to a literal of sufficient size or you will get an overflow warning as well. I get that functionally it is the same, but compilers are more strict about this by default now.